yt/geometry/selection_routines.pyx

```diff
@@ -1473,10 +1473,19 @@ cdef class SliceSelector(SelectorObject):
     cdef int axis
     cdef np.float64_t coord
     cdef int ax, ay
+    cdef int reduced_dimensionality
 
     def __init__(self, dobj):
         self.axis = dobj.axis
         self.coord = _ensure_code(dobj.coord)
+        # If we have a reduced dimensionality dataset, we want to avoid any
+        # checks against it in the axes that are beyond its dimensionality.
+        # This means that if we have a 2D dataset, *all* slices along z will
+        # select all the zones.
+        if self.axis >= dobj.ds.dimensionality:
```

Comment:
I know folks that really love `x-z` domains and do a lot of `y`-slices :-)

Reply:
So I *think* in that case, we'll have set `geometry = ("cartesian", "xzy")` and so when we slice along coord 2 it will still represent `y`, right?